sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_intel_matrix.asciidoc

```diff
@@ -244,16 +249,42 @@ the out of bounds data (depicted as x here).
 |<------------`Stride`---------->|
 
 Note that joint matrix shape `Rows` and `Cols` represents a sub-block
-of the picture above for each sub-group execution. The out of bounds
-functions checks that the 2d block starting at the address to be
-accessed `[0, src]` and ending at `[Cols, src + Rows * stride]` does
-not exceed the bigger 2d block starting at `[0, src]` and ending at
-`[Width, src + Height * stride]`. For instance, load of `sub_b`
-defined as `joint_matrix<sub_group, bfloat16, use::b, 2, 3,
-layout::row_major> sub_b;` from `ptr + 9` checks that `ptr + 9` does
-not exceed `ptr + 10` where 10 is `Width` so it reads only 1 element
-for that iteration. The 2 columns left that exceed the total `10`
-columns are filled by zero.
+of the picture above. The out of bounds data results when the global
+matrix size is not evenly divisible by the joint matrix size.
+
+==== Checked APIs
+`joint_matrix_load`, `joint_matrix_store`, and `joint_matrix_fill`
+operations do not perform bounds checking. When the global matrix size is
+not multiple of the joint matrix size, this extension adds a new form
+of API to load, store, and fill joint matrix while checking the
+bounds. For load and fill, the out-of-bounds elements are set to
+0. For the store, they are dropped out.
```

Comment:
Your description looks better, I will change to it. 
However, I don't prefer to use the word "padding" as this may suggest padding occurred (by the user or compiler) which is not necessarily the case, these data are just there in memory. So I would use the word "extra".

yes, the invalid address description is correct as these functions do not try to access the extra elements.